include/cocaine/rpc/dispatch.hpp

```diff
@@ -62,8 +64,21 @@ class basic_dispatch_t {
     // an empty optional - recurrent transition (i.e. no transition at all).
 
     virtual
-    boost::optional<dispatch_ptr_t>
-    process(const decoder_t::message_type& message, const upstream_ptr_t& upstream) const = 0;
+    auto
+    process(int id, const rpc_t& rpc) const -> boost::optional<dispatch_ptr_t> = 0;
+
+    // Observers
+
+    auto
+    name() const -> std::string;
```

Comment:
I think that 'name (input_types_tuple) -> (output_types_tuple)' is how all function declarations should look in all PLs because it's streamlined and makes sense. This is as close as it gets in C++.